File: lessons/02-topics-publishers/cpp/robot_news_station.cpp
#include "rclcpp/rclcpp.hpp"
#include "example_interfaces/msg/string.hpp" //get String message type

class RobotNewsStation:public rclcpp::Node{
    public:
        RobotNewsStation():Node("robot_news_station"){
            //Create a publisher with <MessageType>(topic_name, queue_size)
            publisher_=this->create_publisher<example_interfaces::msg::String>("robot_news",10);
            //create a timer that calls publishNews every second
            timer_=this->create_wall_timer(std::chrono::seconds(1),
            std::bind(&RobotNewsStation::publishNews,this));
            RCLCPP_INFO(this->get_logger(),"RObot News Station has started");//print startup messge onto the console
        }
    private:
        void publishNews() {
            auto msg = example_interfaces::msg::String();//create a string message object
            msg.data="Breaking News: Robot News Station is live!";//fill data field of the message 
            publisher_->publish(msg);//publish the message on the topic
        }
        rclcpp::Publisher<example_interfaces::msg::String>::SharedPtr publisher_;//Shared pointer to a publisher object that can publish String 
        rclcpp::TimerBase::SharedPtr timer_;//Shared pointer to a timer object
};      

int main(int argc, char **argv) {
    rclcpp::init(argc,argv);//start ROS2 communication
    auto node = std::make_shared<RobotNewsStation>();//create an instance of the node
    rclcpp::spin(node);//keep the node alive and processing callbacks
    rclcpp::shutdown();//shutdown ROS2 comunication
    return 0;

}

File: lessons/02-topics-publishers/python/robot_news_station.py
#!/usr/bin/env python3
import rclpy
from rclpy.node import Node
from example_interfaces.msg import String #importing the String message type from example_interfaces package

class RobotNewsStation(Node):
    def __init__(self):
        super().__init__("robot_news_station")
        self.publisher_=self.create_publisher(String,"robot_news",10) #datatype, topic name and queue size for buffer
        self.timer_=self.create_timer(0.5,self.publish_news) #timer to call the publish_news function every 0.5 seconds
        self.get_logger().info("Robot News Station has been started.")#Print startup message onto the console

    def publish_news(self):
        msg=String()#Create string message object
        msg.data="Hello, this is the latest news from the robot news station!"#Fill message data field
        self.publisher_.publish(msg)#Publish message on topic

def main(args=None):
    rclpy.init(args=args)#Initialize ROS2 communication
    node=RobotNewsStation()#Create node object
    rclpy.spin(node)#Keep node alive and process callbacks
    rclpy.shutdown()#Shutdown ROS2 communication when done 

if __name__=="__main__":
    main()